### ISSUE TITLE
Environment variables in setapikey

### ISSUE THREAD

**Issue by CJHarmath** [NuGet/Home#7145]
## Details about Problem
NuGet product used: NuGet.exe
NuGet version 4.6.2.5055

When using an environment variable with setapikey, it's not getting evaluated during set or during push.

## Use case
Use case is kind of weird, but here it goes:
We have a custom Nuget Server implementation which emails the Kerberos authenticated user as the package pusher which works great when humans use it.
But when the CI server running as a service account is pushing then you can only see the CI server's account as the authenticated user and I got a request to try to include the git committer as well.

To workaround this, I've hijacked the API Key (as haven't seen any other ways to add a custom header to nuget push) to include the git author which is injected as an env var on my Jenkins instance.
So it looks something like this:
nuget push *.nupkg -source MyNugetServer -apiKey OnBehalfOf:%gitlabUserEmail%

And on the server side, I just check the X-NuGet-ApiKey if it has the on behalf of.
The actual auth(z) is done via Kerberos / AD group membership, so the API key is ignored otherwise.

Based on the docs [here](https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#environment-variables-in-configuration) I wasn't sure if it's supported or not, but it seems like it's not working, so just wanted to clarify if this something which would make sense to support ?
Or if you guys have a better idea to pass some extra header with nuget.exe push ? ( which TBH would be a more elegant solution )

## Detailed repro steps so we can see the same problem

1. PS> $env:NugetApiKey = "12345"
1. nuget setapikey "%NugetApiKey%" -source test
1. nuget push *.nupkg -source test

Result: API key is passed in as "%NugetApiKey%" instead of 12345

Thanks for checking!

**Comment by nkolev92** [comment]
It's not supported currently.
The nuget api key would need to be associated with a source however so it can't be as simple as the above.

**Comment by CJHarmath** [comment]
Thanks @nkolev92 for checking!
I figured it's not supported right now as the value seems to be encrypted in the config after calling setapikey.

A more generalized way to pass extra information from nuget.exe to a nuget service could be allowing custom headers to be passed in via command line args or config per source. 
Similarly to the X-NuGet-ApiKey.

For example:
`nuget push *.nupkg -source test -header "X-NuGet-OnBehalfOf:joe"`

Allowing use cases such as a CI server adding a header with the git commit author, etc.

Thanks

**Comment by nkolev92** [comment]
Related to what you are articulating. 
https://github.com/NuGet/Home/issues/6574

That issue talks about a plugin that can add params to the http requests.

**Comment by CJHarmath** [comment]
yeah, something like that could also be surfaced as a command line arg / config attribute per source.
Being able to use environment variables as header values per source would be an extra nice touch for the CI server use-case as for example Jenkins already sets a bunch of GIT releated env vars.

**Comment by Frulfump** [comment]
Why does the bot close it as completed?
